text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+### Tranche 1 - Base functionality, leaf nodes in many dependency trees
+* ‘syn’ equivalent
+* ‘bitflags’ equivalent
+* ‘quote’ equivalent
+* Platform bindings (libc)
+* Encoding support (base64, hex, etc)
+* ‘cfg-if’ equivalent
+* Random number generation
+* Serialization/deserialization
+* Data structure serialization
+* Date/time support
+* Error handling support
+
+### Tranche 2: Next step up in complexity/abstraction
+* Regular expressions
+* Compression (e.g. zlib, zstd)
+* Command line argument parsing
+* Tracing
+
+### Tranche 3 - Common higher level application building blocks
+* Default asynchronous runtime
+* Cryptography
+* Zeroize memory
+* TLS
+* HTTP
+
+Ultimately the offered functionality would be selected by the ESL community.
+
+The ESL can grow at a pace that is comfortable given available resources. This means that it might take a year or two to advance to the next tranche of functionality.
+
+It should be OK to remove previously included functionality if doing so makes sense. This is part of evolving APIs, and should be regarded as potentially healthy as opposed to necessarily reflective of a failure.
+
+## Adding Functionality
+
+Once a decision is made to try to include functionality in the ESL, the ESL community should try to find a suitable existing crate with maintainers who are willing to join the ESL.
+
+If a maintainer chooses to move their crate into the ESL, doing so does not require them to relinquish their role as a maintainer. They do need to commit to upholding the policies that the ESL community has decided on. In order to help with that and other maintenance tasks, maintainers will receive various forms of support as part of ESL membership (see below).
+
+If the ESL community would like to add functionality but is unable to find a suitable crate with maintainers willing to join, the community will have to consider their options, which include creating a new crate or not adding the functionality. If the functionality is required in order to enable the addition of higher-order functionality, an exception could be made to allow for a dependency outside of the ESL.
+
+## Maintainer Support
+
+The ESL should endeavor to provide the following forms of support to maintainers:
+
+* **Infrastructure Assistance**. Testing, security, release, and benchmarking infrastructure managed by an ESL infrastructure partner.
+* **Funding**. The opportunity to receive grants to help cover the cost of their contributions.
+* **Security Release Coverage**. If maintainers are unavailable to respond to a security issue, there is a team that can back them up.
+* **Legal Assistance**. Legal assistance as necessary from an ESL legal partner.
+
+## Crate Names
+
+All crates in the ESL would ideally have names that:
+
+1) Clearly describe the functionality they offer
+2) Contain a reliable indicator of ESL membership
```

Comment:
This should be out of scope. Renames are a breaking change and requiring one to add/remove branding is a bad idea.

Reply:
In theory cargo/crates.io could support "redirect" renames. It's a thing that was briefly discussed during the namespacing RFC, it didn't end up happening, but it's ... something that could be proposed.

I agree that _without_ such a change it's probably not going to work.

Reply:
We should not count on being able to do so though as I expect some tricky problems in the resolver and Cargo's ideals around ‘Cargo.lock` files.